src/components/GameField/index.tsx

```diff
@@ -0,0 +1,159 @@
+  }
+
+  componentDidUpdate = (prevProps: GameFieldProps) => {
+    if ((prevProps.fieldSize !== this.props.fieldSize)
+      || (prevProps.ballsCount !== this.props.ballsCount)) {
+      if (this.timerId) {
+        clearInterval(this.timerId);
+      }
+    }
+  }
+
+  onBallClickHandler = (
+    event: React.MouseEvent<HTMLImageElement, MouseEvent>,
+    rowIndex: number,
+    cellIndex: number,
+    colorCode: number
+  ): void => {
+    document.querySelectorAll('.jumping').forEach((elem) => elem.classList.remove('jumping'));
+    (event.target as HTMLImageElement).classList.add('jumping');
+    if (this.timerId) {
+      clearInterval(this.timerId);
+    }
+
+    this.playSound();
+
+    this.timerId = setInterval(() => {
+      this.playSound()
+    }, 1000);
+
+    this.setState(() => ({
+      currentActive: {
+        row: rowIndex,
+        column: cellIndex,
+        colorIndex: colorCode,
+      }
+    }));
+    event.stopPropagation();
+  }
+
+  playSound = (): void => {
+    const { playSound } = this.props;
+    if (!playSound) {
+      return;
+    }
+    const playPromise = this.audioRef.current?.play();
+    if (playPromise !== undefined) {
+      playPromise.then(function () {
+        //do play sound
+      }).catch(function (error) {
+      });
+    }
+  }
+
+  onCellClickHandler = (rowIndex: number, cellIndex: number): void => {
+    const { moveBallToNewCell } = this.props;
+    const { currentActive } = this.state;
+    if (!currentActive) {
+      return;
+    }
+    const path = this.checkTargetCellAvailable(rowIndex, cellIndex);
+    if (path != null) {
+      moveBallToNewCell(currentActive, rowIndex, cellIndex, path);
+      document.querySelectorAll('.jumping').forEach((elem) => elem.classList.remove('jumping'));
+      clearInterval(this.timerId);
+      this.setState({ currentActive: null });
+    }
+  }
+
+  checkTargetCellAvailable = (rowIndex: number, cellIndex: number): [number, number][] => {
+    const { gameFieldState, fieldSize } = this.props;
+    const { currentActive } = this.state;
+    return gameFieldState[rowIndex] && gameFieldState[rowIndex][cellIndex] === 0
+      && findPathToTarget(gameFieldState, fieldSize, [currentActive.row, currentActive.column], [rowIndex, cellIndex]);
+  }
+
+  getTableCells = (rowIndex: number): JSX.Element[] => {
+    const { gameFieldState, fieldSize } = this.props;
+    const result: JSX.Element[] = [];
+    for (let i = 0; i < fieldSize; i += 1) {
+      const ballCode = gameFieldState[rowIndex] && gameFieldState[rowIndex][i];
+      const ball = ballCode > 0 && ballCode || null;
+      result.push(<td className="tableCell" key={`${rowIndex.toString()}${i.toString()}`} onClick={() => this.onCellClickHandler(rowIndex, i)}>
+        {
+          ball !== null && this.imagesCache[ball - 1] && (<img className="ball-image" src={this.imagesCache[ball - 1].src} alt="" onClick={(event) => this.onBallClickHandler(event, rowIndex, i, ball)}></img>
```

Comment:
Think about eslint rule to cut code line length.
Code lines like this really hard to understand.
Your code length should be not more than 100 symbols.